core/src/main/kotlin/com/strumenta/kolasu/cli/ASTProcessingCommand.kt

```diff
@@ -0,0 +1,134 @@
+package com.strumenta.kolasu.cli
+
+import com.github.ajalt.clikt.core.CliktCommand
+import com.github.ajalt.clikt.parameters.arguments.argument
+import com.github.ajalt.clikt.parameters.arguments.multiple
+import com.github.ajalt.clikt.parameters.options.default
+import com.github.ajalt.clikt.parameters.options.flag
+import com.github.ajalt.clikt.parameters.options.help
+import com.github.ajalt.clikt.parameters.options.option
+import com.github.ajalt.clikt.parameters.types.file
+import com.strumenta.kolasu.model.Node
+import com.strumenta.kolasu.parsing.ASTParser
+import com.strumenta.kolasu.parsing.ParsingResult
+import com.strumenta.kolasu.validation.IssueSeverity
+import com.strumenta.kolasu.validation.Result
+import java.io.File
+import java.nio.charset.Charset
+import java.util.function.Function
+import kotlin.system.exitProcess
+
+typealias ParserInstantiator<P> = Function<File, P?>
+
+abstract class ASTProcessingCommand<R : Node, P : ASTParser<R>>(
```

Comment:
The idea is to have this base class for all commands that could process entire directories containing source code and produce a sequence of ASTs, each to be processed